RoT/Protocol/RFCs/0001-Transport_Agnosticism.md

```diff
@@ -0,0 +1,149 @@
+   sized byffer of bytes) from one (not necesarily addressable) endpoint to
+   another. In other words, the admissible transport is responsible for removing
+   frames from packets and assembling them in sequence.
+
+2. Each endpoint has a known *maximum message length*, measured in bytes. The
+   transport is responsible for negotiating this parameter for each device a
+   Cerberus device intends to speak to. This parameter must be made available
+   to the Cerberus protocol but Cerberus itself does not negotiate it. This
+   paramter must be at least 64 bytes.
+
+3. Each message is either the request or response half of a Cerberus command, as
+   defined in the Challenge Protocol. To uniquely identify the type of an
+   incoming message, the binding of Cerberus to the admissible transport must
+   specify a *transport-specific header* that contains at a minimum the folling
+   parameters. How they are encoded is irrelevant to Cerberus, beyond that they
+   be computable without recieving the entire incoming message:
+   - The command type byte (as specified in the Challenge Protocol).
+   - A single bit: whther this message is the request or response half of the
+     command.
+   - The length of the incoming message, in bytes.
+   - Addressing information that determines which device sent the message
+     (the actual contents of this information is irrelevant to Cerberus).
+
+4. Messages are "addressed", in two respects:
+   a. A server can take a request and reply to the original sender with a
+      response, and a client can match up requests and their responses.
+   b. Cerberus can use addressing information of unspecified format to
+      construct requests to a specific device (this opaque addressing
+      information could be present in a PCD, for example).
+
+By construction, MCTP almost fulills the above requirements: it provides framing
+of a message of arbitrary size, and the PCIe Vendor command provides a location
+for the transport-specific header. Note that Cerberus itself performs the
+size negotiation at the moment, so that would be moved out of the Challenge
+Protocol and into the transport layer.
+
+The above list is everything Cerberus needs: Cerberus can be spoken over any
+admissible transport with not change to the actual bytes present in the
+encoded messages.
+
+# Specification Changelist
+
+This RFC seeks the following changes to the Challenge Protocol spec:
+
+- Chapter 3 (Protocol and Hierarchy) should be replaced with the list of
+  requirements given above. We do not recommend using the above text
+  exactly; instead, the new Chapter 3 should carefully elaborate each point,
+  to make it possible for third parties to evaluate whether their chosen
+  transport layer is admissible.
+
+- The current contents of Chapter 3 should be moved into an appendix, provided
+  as an example admissible transport layer for I2C connections. The same should
```

Comment:
s/transport layer for I2C connections/transport layer for SMBus/MCTP connections/g ?

Reply:
Well, no. MCTP *is* the transport, and I2C is the bus it runs on.

Reply:
SMBus would be the bus though